lib/streamlit/elements/widgets/multiselect.py

```diff
@@ -62,24 +69,47 @@
     )
 
 
-@dataclass
 class MultiSelectSerde(Generic[T]):
     options: Sequence[T]
-    default_value: list[int] = field(default_factory=list)
+    formatted_option_to_option_mapping: dict[str, T]
+    default_options_indices: list[int]
+    format_func: Callable[[Any], str]
+    accept_new_options: bool
 
-    def serialize(self, value: list[T]) -> list[int]:
-        indices = check_and_convert_to_indices(self.options, value)
-        return indices if indices is not None else []
+    def __init__(
+        self,
+        options: Sequence[T],
+        formatted_option_to_option_mapping: dict[str, T],
+        default_options_indices: list[int] | None = None,
+        format_func: Callable[[Any], str] = str,
+        accept_new_options: bool = False,
+    ):
+        self.options = options
+        self.formatted_option_to_option_mapping = formatted_option_to_option_mapping
+        self.default_options_indices = default_options_indices or []
+        self.accept_new_options = accept_new_options
+        self.format_func = format_func
+
+    def serialize(self, value: list[T | str] | list[T]) -> list[str]:
+        return [self.format_func(v) for v in value] if value is not None else []
```

Comment:
Same comment as in selectbox, do we really need to `format_func` here since it could be a bit risky.

Reply:
I replied in the other [comment thread](https://github.com/streamlit/streamlit/pull/10788#discussion_r2013612695).